code/modules/shuttle/custom_shuttle.dm

```diff
@@ -0,0 +1,268 @@
+	linkedShuttle.port_direction = angle2dir(dir2angle(dir) - (dir2angle(linkedShuttle.dir)) + 180)
+	linkedShuttle.preferred_direction = NORTH
+	linkedShuttle.ignitionTime = CUSTOM_ENGINES_START_TIME
+	linkedShuttle.count_engines()
+	linkedShuttle.hyperspace_sound(HYPERSPACE_WARMUP)
+	var/throwForce = clamp((calculated_speed / 2) - 5, 0, 10)
+	linkedShuttle.movement_force = list("KNOCKDOWN" = calculated_speed > 5 ? 3 : 0, "THROW" = throwForce)
+	if(!(targetLocation in params2list(possible_destinations)))
+		log_admin("[usr] attempted to launch a shuttle that has been affected by href dock exploit on [src] with target location \"[targetLocation]\"")
+		message_admins("[usr] attempted to launch a shuttle that has been affected by href dock exploit on [src] with target location \"[targetLocation]\"")
+		return
+	switch(SSshuttle.moveShuttle(shuttleId, targetLocation, 1))
+		if(0)
+			consumeFuel(dist)
+			say("Shuttle departing. Please stand away from the doors.")
+		if(1)
+			to_chat(usr, "<span class='warning'>Invalid shuttle requested.</span>")
+		else
+			to_chat(usr, "<span class='notice'>Unable to comply.</span>")
+	return
+
+/obj/machinery/computer/custom_shuttle/connect_to_shuttle(obj/docking_port/mobile/port, obj/docking_port/stationary/dock, idnum, override=FALSE)
+	if(port && (shuttleId == initial(shuttleId) || override))
+		linkShuttle(port.id)
+
+//Custom shuttle docker locations
+/obj/machinery/computer/camera_advanced/shuttle_docker/custom
+	name = "Shuttle Navigation Computer"
+	desc = "Used to designate a precise transit location for private ships."
+	lock_override = NONE
+	whitelist_turfs = list(/turf/open/space,
+		/turf/open/lava,
+		/turf/open/floor/plating/beach,
+		/turf/open/floor/plating/ashplanet,
+		/turf/open/floor/plating/asteroid,
+		/turf/open/floor/plating/lavaland_baseturf)
+	jumpto_ports = list("whiteship_home" = 1)
+	view_range = 12
+	designate_time = 100
+	circuit = /obj/item/circuitboard/computer/shuttle/docker
+
+/obj/machinery/computer/camera_advanced/shuttle_docker/custom/Initialize()
+	. = ..()
+	GLOB.jam_on_wardec += src
+
+/obj/machinery/computer/camera_advanced/shuttle_docker/custom/Destroy()
+	GLOB.jam_on_wardec -= src
+	return ..()
+
+/obj/machinery/computer/camera_advanced/shuttle_docker/custom/placeLandingSpot()
+	if(!shuttleId)
+		return	//Only way this would happen is if someone else delinks the console while in use somehow
+	var/obj/docking_port/mobile/M = SSshuttle.getShuttle(shuttleId)
+	if(M?.mode != SHUTTLE_IDLE)
+		to_chat(usr, "<span class='warning'>You cannot target locations while in transit.</span>")
+		return
+	..()
+
+/obj/machinery/computer/camera_advanced/shuttle_docker/custom/attack_hand(mob/user)
+	if(!shuttleId)
+		to_chat(user, "<span class='warning'>You must link the console to a shuttle first.</span>")
+		return
+	return ..()
+
+/obj/machinery/computer/camera_advanced/shuttle_docker/custom/proc/linkShuttle(var/new_id)
+	shuttleId = new_id
+	shuttlePortId = "shuttle[new_id]_custom"
```

Comment:
This used to break on bee if you make more than 1 shuttle docker so might want to check it out

Reply:
> Should be noted that the engines and heaters cause hard dels due to not doing anything when deleted.

Sorry for taking so long to reply, I got burnt out from working on this for a while and just now got back to working on it. I talked with Jamie about this and he said that its not something to worry about.

Reply:
> This used to break on bee if you make more than 1 shuttle docker so might want to check it out

Was there a PR that fixed it, or did you instead just rework the shuttle system to get around it?

Reply:
It was fixed as a part of the shuttle system rework

Reply:
I think the cause was the port ID not being unique for each console. Just adding a static number and sticking _[varname++] on the end should fix it I think

Reply:
It was either that or the ports weren't being deleted properly.